File: frontend/src/services/resource.service.ts
/*
 * Copyright 2025 Aetheris RAG Team. All rights reserved.
 */

import api from './api'

/**
 * 资源上传请求
 */
export interface ResourceUploadRequest {
  filePath: string
  title: string
  tags?: string
  description?: string
}

/**
 * 资源信息
 */
export interface Resource {
  id: number
  title: string
  tags?: string
  fileType: string
  fileSize: number
  description?: string
  contentHash: string
  uploadedBy: number
  uploadTime: string
  chunkCount: number
  vectorized: boolean
  duplicate?: boolean // 是否为重复上传
}

/**
 * 切片信息
 */
export interface Chunk {
  id: number
  resourceId: number
  chunkIndex: number
  chunkText: string
  locationInfo?: string
  pageStart?: number
  pageEnd?: number
  chapterPath?: string
  vectorized: boolean
  createdAt: string
}

/**
 * 分页响应
 */
export interface PageResponse<T> {
  items: T[]
  total: number
  page: number
  size: number
  totalPages: number
  hasNext: boolean
}

/**
 * 资源服务类
 */
export class ResourceService {
  /**
   * 上传资源
   */
  static async upload(data: ResourceUploadRequest): Promise<Resource> {
    const response = await api.post<any, Resource>('/resources', data)
    return response
  }

  /**
   * 获取资源列表（分页）
   */
  static async getResourceList(page = 0, size = 10): Promise<PageResponse<Resource>> {
    const response = await api.get<any, PageResponse<Resource>>('/resources', {
      params: { page, size }
    })
    return response
  }

  /**
   * 获取资源详情
   */
  static async getResourceById(id: number): Promise<Resource> {
    const response = await api.get<any, Resource>(`/resources/${id}`)
    return response
  }

  /**
   * 获取资源切片列表
   */
  static async getChunksByResourceId(id: number): Promise<Chunk[]> {
    const response = await api.get<any, Chunk[]>(`/resources/${id}/chunks`)
    return response
  }

  /**
   * 更新资源信息
   */
  static async updateResource(id: number, data: {
    title: string
    tags?: string
    description?: string
  }): Promise<Resource> {
    const response = await api.put<any, Resource>(`/resources/${id}`, data)
    return response
  }

  /**
   * 删除单个资源（用于资源详情页）。
   */
  static async deleteResource(id: number): Promise<void> {
    await api.delete(`/resources/${id}`)
  }

  /**
   * 批量删除资源（用于资源列表页）。
   * 单个删除时也使用此接口，传入 ids=[id]
   */
  static async batchDeleteResources(ids: number[]): Promise<void> {
    await api.delete('/resources/batch', { data: { ids } })
  }
}

export default ResourceService

File: backend/src/main/java/com/aetheris/rag/service/ResourceService.java
/*
 * Copyright 2025 Aetheris RAG Team. All rights reserved.
 */
package com.aetheris.rag.service;

import com.aetheris.rag.entity.Chunk;
import com.aetheris.rag.entity.Resource;
import java.util.List;

/**
 * 资源服务接口。
 *
 * <p>提供资源查询、更新等CRUD功能。
 * 文档处理和向量化功能已移至 ProcessingService。
 *
 * @author Aetheris Team
 * @version 2.0.0
 * @since 2025-12-30
 */
public interface ResourceService {

  /**
   * 根据ID查询资源。
   *
   * @param id 资源ID
   * @return 资源实体，不存在则返回 null
   */
  Resource getResourceById(Long id);

  /**
   * 查询资源列表（分页）。
   *
   * @param offset 偏移量
   * @param limit 限制数量
   * @return 资源列表
   */
  List<Resource> getResourceList(int offset, int limit);

  /**
   * 查询资源的所有切片。
   *
   * @param resourceId 资源ID
   * @return 切片列表
   */
  List<Chunk> getChunksByResourceId(Long resourceId);

  /**
   * 检查资源是否存在（根据内容哈希）。
   *
   * @param contentHash 内容哈希（SHA-256）
   * @return 资源实体，不存在则返回 null
   */
  Resource findByContentHash(String contentHash);

  /**
   * 统计资源总数。
   *
   * @return 资源总数
   */
  Long getResourceCount();

  /**
   * 更新资源信息。
   *
   * @param id 资源ID
   * @param title 标题
   * @param tags 标签
   * @param description 描述
   * @return 更新后的资源
   */
  Resource updateResource(Long id, String title, String tags, String description);

  /**
   * 更新资源的向量化状态。
   *
   * <p>供 VectorService 在向量化完成后回调使用。</p>
   *
   * @param resourceId 资源ID
   * @param vectorized 是否已向量化
   */
  void updateVectorizationStatus(Long resourceId, boolean vectorized);

  /**
   * 更新资源的切片数量和向量化状态。
   *
   * <p>供 VectorService 在向量化完成后回调使用。</p>
   *
   * @param resourceId 资源ID
   * @param chunkCount 切片数量
   * @param vectorized 是否已向量化
   */
  void updateChunkVectorizationStatus(Long resourceId, Integer chunkCount, boolean vectorized);
}
